core/src/com/mygdx/game/models/HexBoardClass.java

```diff
@@ -0,0 +1,113 @@
+package com.mygdx.game.models;
+
```

Comment:
Everytime you use object you should just be using T. When u use object your saying that any object could be passed in. A single hex board could have booleans, strings, or integers. But we want a hex Board to have a single type T. When someone declares a hesboard with a Generic they write. "HexBoard String hexboard = new HexBoard();" The "String" part of that statement makes every thing T a String. For examplle T getHex() would be String getHex.

Reply:
http://www.tutorialsteacher.com/Content/images/csharp/generic-class.png